aggregator_api/src/lib.rs

```diff
@@ -435,12 +458,153 @@ async fn get_task_metrics<C: Clock>(
     }))
 }
 
+async fn get_global_hpke_keypairs<C: Clock>(
+    _: &mut Conn,
+    State(ds): State<Arc<Datastore<C>>>,
+) -> Result<impl Handler, Status> {
+    Ok(Json(
+        ds.run_tx_with_name("get_global_hpke_keypairs", |tx| {
+            Box::pin(async move { tx.get_global_hpke_keypairs().await })
+        })
+        .await
+        .map_err(|err| {
+            error!(err = %err, "Database transaction error");
+            Status::InternalServerError
+        })?
+        .into_iter()
+        .map(GlobalHpkeKeypairResp::from)
+        .collect::<Vec<_>>(),
+    ))
+}
+
+async fn get_global_hpke_keypair<C: Clock>(
+    conn: &mut Conn,
+    State(ds): State<Arc<Datastore<C>>>,
+) -> Result<impl Handler, Status> {
+    let config_id = conn.hpke_config_id_param()?;
+    Ok(Json(GlobalHpkeKeypairResp::from(
+        ds.run_tx_with_name("get_global_hpke_keypair", |tx| {
+            Box::pin(async move { tx.get_global_hpke_keypair(&config_id).await })
+        })
+        .await
+        .map_err(|err| {
+            error!(err = %err, "Database transaction error");
+            Status::InternalServerError
+        })?
+        .ok_or(Status::NotFound)?,
+    )))
+}
+
+async fn put_global_hpke_keypair<C: Clock>(
+    _: &mut Conn,
+    (State(ds), Json(req)): (State<Arc<Datastore<C>>>, Json<PutGlobalHpkeKeypairReq>),
+) -> Result<impl Handler, Status> {
+    let existing_keypairs = ds
+        .run_tx_with_name("put_global_hpke_keypair_determine_id", |tx| {
+            Box::pin(async move { tx.get_global_hpke_keypairs().await })
+        })
+        .await
+        .map_err(|err| {
+            error!(err = %err, "Database transaction error");
+            Status::InternalServerError
+        })?
+        .iter()
+        .map(|keypair| u8::from(*keypair.hpke_keypair().config().id()))
+        .collect::<Vec<_>>();
```

Comment:
With the design of the global keys, we have to keep them around for some time before we can fully delete them from the database. The reason is clients cache the HPKE config, so they might upload reports encrypted with a key that we consider expired. So, if we're at active config 255 with [0,254] free, then we have to wrap around... but there's no guarantee that ID 0 isn't taken because deletion is human driven.

If key management was 100% automated, this wouldn't be a problem, but there's a human element at the moment that can cause non-contiguous caps in [0,254]. Thus I think the algorithm has to be "the first available integer in [0,255]" where that interval has potentially non-contiguous gaps.